crates/storage/codecs/src/lib.rs

```diff
@@ -479,6 +486,104 @@ const fn decode_varuint_panic() -> ! {
     panic!("could not decode varuint");
 }
 
+/// Trait that will transform the data to be saved in the DB.
+pub trait Encode: Send + Sync + Sized + Debug {
```

Comment:
can we move this to a different mod, otherwise this makes it harder to navigate this crate